diff -urN a/extra-lib.mk b/extra-lib.mk
--- a/extra-lib.mk	2014-04-26 21:46:57.650986900 +0100
+++ b/extra-lib.mk	2014-04-26 21:47:34.849114500 +0100
@@ -13,7 +13,7 @@
 
 ifneq (,$($(lib)-static-only-routines))
 ifneq (,$(filter yes%,$(build-shared)$($(lib).so-version)))
-object-suffixes-$(lib) += $(filter-out $($(lib)-inhibit-o),.oS)
+object-suffixes-$(lib) += $(filter-out $($(lib)-inhibit-o),.oSTATIC)
 endif
 endif
 
@@ -29,7 +29,7 @@
 
 # Add each flavor of library to the lists of things to build and install.
 install-lib += $(foreach o,$(object-suffixes-$(lib)),$(lib:lib%=$(libtype$o)))
-extra-objs += $(foreach o,$(filter-out .os .oS,$(object-suffixes-$(lib))),\
+extra-objs += $(foreach o,$(filter-out .os .oSTATIC,$(object-suffixes-$(lib))),\
 			$(patsubst %,%$o,$(filter-out \
 					   $($(lib)-shared-only-routines),\
 					   $(all-$(lib)-routines))))
@@ -37,8 +37,8 @@
 extra-objs += $(patsubst %,%.os,$(filter-out $($(lib)-static-only-routines),\
 					     $(all-$(lib)-routines)))
 endif
-ifneq (,$(filter .oS,$(object-suffixes-$(lib))))
-extra-objs += $(patsubst %,%.oS,$(filter $($(lib)-static-only-routines),\
+ifneq (,$(filter .oSTATIC,$(object-suffixes-$(lib))))
+extra-objs += $(patsubst %,%.oSTATIC,$(filter $($(lib)-static-only-routines),\
 					 $(all-$(lib)-routines)))
 endif
 alltypes-$(lib) := $(foreach o,$(object-suffixes-$(lib)),\
@@ -62,7 +62,7 @@
 
 
 # Use o-iterator.mk to generate a rule for each flavor of library.
-ifneq (,$(filter-out .os .oS,$(object-suffixes-$(lib))))
+ifneq (,$(filter-out .os .oSTATIC,$(object-suffixes-$(lib))))
 define o-iterator-doit
 $(objpfx)$(patsubst %,$(libtype$o),$(lib:lib%=%)): \
   $(patsubst %,$(objpfx)%$o,\
@@ -70,7 +70,7 @@
 			  $(all-$(lib)-routines))); \
 	$$(build-extra-lib)
 endef
-object-suffixes-left = $(filter-out .os .oS,$(object-suffixes-$(lib)))
+object-suffixes-left = $(filter-out .os .oSTATIC,$(object-suffixes-$(lib)))
 include $(patsubst %,$(..)o-iterator.mk,$(object-suffixes-left))
 endif
 
@@ -82,9 +82,9 @@
 	$(build-extra-lib)
 endif
 
-ifneq (,$(filter .oS,$(object-suffixes-$(lib))))
-$(objpfx)$(patsubst %,$(libtype.oS),$(lib:lib%=%)): \
-  $(patsubst %,$(objpfx)%.oS,\
+ifneq (,$(filter .oSTATIC,$(object-suffixes-$(lib))))
+$(objpfx)$(patsubst %,$(libtype.oSTATIC),$(lib:lib%=%)): \
+  $(patsubst %,$(objpfx)%.oSTATIC,\
 	     $(filter $($(lib)-static-only-routines),\
 		      $(all-$(lib)-routines)))
 	$(build-extra-lib)
diff -urN a/Makeconfig b/Makeconfig
--- a/Makeconfig	2014-04-26 21:46:58.170016600 +0100
+++ b/Makeconfig	2014-04-26 21:47:34.856114900 +0100
@@ -469,7 +469,7 @@
 link-libc-tests-rpath-link = $(link-libc-rpath-link)
 endif
 link-libc-before-gnulib = $(common-objpfx)libc.so$(libc.so-version) \
-			  $(common-objpfx)$(patsubst %,$(libtype.oS),c) \
+			  $(common-objpfx)$(patsubst %,$(libtype.oSTATIC),c) \
 			  $(as-needed) $(common-objpfx)elf/ld.so \
 			  $(no-as-needed)
 link-libc = $(link-libc-rpath-link) $(link-libc-before-gnulib) $(gnulib)
@@ -813,7 +813,7 @@
 # The compilation rules use $(CPPFLAGS-${SUFFIX}) and $(CFLAGS-${SUFFIX})
 # to pass different flags for each flavor.
 libtypes = $(foreach o,$(object-suffixes-for-libc),$(libtype$o))
-all-object-suffixes := .o .os .op .og .oS
+all-object-suffixes := .o .os .op .og .oSTATIC
 object-suffixes :=
 CPPFLAGS-.o = $(pic-default)
 CFLAGS-.o = $(filter %frame-pointer,$(+cflags))
@@ -852,14 +852,14 @@
 
 ifeq (yes,$(build-shared))
 # Build special library that contains the static-only routines for libc.
-object-suffixes-for-libc += .oS
+object-suffixes-for-libc += .oSTATIC
 
 # Must build the routines as PIC, though, because they can end up in (users')
 # shared objects.  We don't want to use CFLAGS-os because users may, for
 # example, make that processor-specific.
-CFLAGS-.oS = $(CFLAGS-.o) $(PIC-ccflag)
-CPPFLAGS-.oS = $(CPPFLAGS-.o) -DPIC -DLIBC_NONSHARED=1
-libtype.oS = lib%_nonshared.a
+CFLAGS-.oSTATIC = $(CFLAGS-.o) $(PIC-ccflag)
+CPPFLAGS-.oSTATIC = $(CPPFLAGS-.o) -DPIC -DLIBC_NONSHARED=1
+libtype.oSTATIC = lib%_nonshared.a
 endif
 
 # The assembler can generate debug information too.
diff -urN a/Makerules b/Makerules
--- a/Makerules	2014-04-26 21:46:58.171016700 +0100
+++ b/Makerules	2014-04-26 21:47:34.859115100 +0100
@@ -161,14 +161,14 @@
 # We only generate these in the top-level makefile, to avoid any weirdness
 # from subdir-specific makefile tweaks creeping in on an update.
 $(common-objpfx)%.make: $(..)%.make.c $(..)Makerules $(common-before-compile)
-	rm -f $@T $@.dT
+	rm -f $@T $@.dTEMP
 	(echo '# Generated from $*.make.c by Makerules.'; \
 	 $(CC) $(CFLAGS) $(CPPFLAGS) -E -DASSEMBLER $< \
-	       -MD -MP -MT '$$(common-objpfx)$*.make' -MF $@.dT \
+	       -MD -MP -MT '$$(common-objpfx)$*.make' -MF $@.dTEMP \
 	 | sed -n '/@@@/{s/@@@[	 ]*\(.*\)@@@/\1/;s/[	 ]*$$//p;}'; \
 	 echo 'common-generated += $(@F)'; \
-	 sed $(sed-remove-objpfx) $(sed-remove-dotdot) $@.dT; \
-	 rm -f $@.dT) > $@T
+	 sed $(sed-remove-objpfx) $(sed-remove-dotdot) $@.dTEMP; \
+	 rm -f $@.dTEMP) > $@T
 	mv -f $@T $@
 endif
 
@@ -327,7 +327,7 @@
 endif # $(build-shared) = yes
 endif # sysd-sorted-done
 
-# Generate .dT files as we compile.
+# Generate .dTEMP files as we compile.
 compile-mkdep-flags = -MD -MP -MF $@.dt -MT $@
 compile-command.S = $(compile.S) $(OUTPUT_OPTION) $(compile-mkdep-flags)
 compile-command.c = $(compile.c) $(OUTPUT_OPTION) $(compile-mkdep-flags)
@@ -404,7 +404,7 @@
 endif
 endif
 
-elide-routines.oS += $(filter-out $(static-only-routines),\
+elide-routines.oSTATIC += $(filter-out $(static-only-routines),\
 				  $(routines) $(aux) $(sysdep_routines))
 elide-routines.os += $(static-only-routines)
 
@@ -678,7 +678,7 @@
 
 +depfiles := $(sources:.c=.d) \
 	     $(patsubst %.o,%.d,$(filter %.o,$(extra-objs:.os=.o))) \
-	     $(patsubst %.oS,%.d,$(filter %.oS,$(extra-objs))) \
+	     $(patsubst %.oSTATIC,%.d,$(filter %.oSTATIC,$(extra-objs))) \
 	     $(patsubst %.o,%.d,$(filter %.o,$(extra-test-objs:.os=.o))) \
 	     $(addsuffix .d,$(tests) $(xtests) $(test-srcs))
 ifeq ($(build-programs),yes)
@@ -969,7 +969,7 @@
 install: $(inst_libdir)/libc.so
 $(inst_libdir)/libc.so: $(common-objpfx)format.lds \
 			$(common-objpfx)libc.so$(libc.so-version) \
-			$(inst_libdir)/$(patsubst %,$(libtype.oS),\
+			$(inst_libdir)/$(patsubst %,$(libtype.oSTATIC),\
 						  $(libprefix)$(libc-name)) \
 			$(+force)
 	(echo '/* GNU ld script';\
@@ -977,7 +977,7 @@
 	 echo '   the static library, so try that secondarily.  */';\
 	 cat $<; \
 	 echo 'GROUP ( $(slibdir)/libc.so$(libc.so-version)' \
-	      '$(libdir)/$(patsubst %,$(libtype.oS),$(libprefix)$(libc-name))'\
+	      '$(libdir)/$(patsubst %,$(libtype.oSTATIC),$(libprefix)$(libc-name))'\
 	      ' AS_NEEDED (' $(rtlddir)/$(rtld-installed-name) ') )' \
 	) > $@.new
 	mv -f $@.new $@
diff -urN a/nptl/Makefile b/nptl/Makefile
--- a/nptl/Makefile	2014-04-26 21:47:00.009121800 +0100
+++ b/nptl/Makefile	2014-04-26 21:47:34.862115300 +0100
@@ -384,7 +384,7 @@
 
 $(inst_libdir)/libpthread.so: $(common-objpfx)format.lds \
 			      $(objpfx)libpthread.so$(libpthread.so-version) \
-			      $(inst_libdir)/$(patsubst %,$(libtype.oS),\
+			      $(inst_libdir)/$(patsubst %,$(libtype.oSTATIC),\
 							$(libprefix)pthread) \
 			      $(+force)
 	(echo '/* GNU ld script';\
@@ -392,7 +392,7 @@
 	 echo '   the static library, so try that secondarily.  */';\
 	 cat $<; \
 	 echo 'GROUP ( $(slibdir)/libpthread.so$(libpthread.so-version)' \
-	      '$(libdir)/$(patsubst %,$(libtype.oS),$(libprefix)pthread)'\
+	      '$(libdir)/$(patsubst %,$(libtype.oSTATIC),$(libprefix)pthread)'\
 	      ')' \
 	) > $@.new
 	mv -f $@.new $@
diff -urN a/sysdeps/i386/i686/Makefile b/sysdeps/i386/i686/Makefile
--- a/sysdeps/i386/i686/Makefile	2014-04-26 21:47:07.665559700 +0100
+++ b/sysdeps/i386/i686/Makefile	2014-04-26 21:47:34.865115500 +0100
@@ -15,11 +15,11 @@
 CFLAGS-.os += -Wa,-mtune=i686
 CFLAGS-.op += -Wa,-mtune=i686
 CFLAGS-.og += -Wa,-mtune=i686
-CFLAGS-.oS += -Wa,-mtune=i686
+CFLAGS-.oSTATIC += -Wa,-mtune=i686
 
 ASFLAGS-.o += -Wa,-mtune=i686
 ASFLAGS-.os += -Wa,-mtune=i686
 ASFLAGS-.op += -Wa,-mtune=i686
 ASFLAGS-.og += -Wa,-mtune=i686
-ASFLAGS-.oS += -Wa,-mtune=i686
+ASFLAGS-.oSTATIC += -Wa,-mtune=i686
 endif
diff -urN a/sysdeps/ieee754/ldbl-opt/Makefile b/sysdeps/ieee754/ldbl-opt/Makefile
--- a/sysdeps/ieee754/ldbl-opt/Makefile	2014-04-26 21:47:07.816568400 +0100
+++ b/sysdeps/ieee754/ldbl-opt/Makefile	2014-04-26 21:47:34.867115600 +0100
@@ -44,7 +44,7 @@
 libnldbl-routines = $(libnldbl-calls:%=nldbl-%)
 libnldbl-inhibit-o = $(object-suffixes)
 libnldbl-static-only-routines = $(libnldbl-routines)
-extra-objs += $(addsuffix .oS, $(libnldbl-routines))
+extra-objs += $(addsuffix .oSTATIC, $(libnldbl-routines))
 
 CFLAGS-nldbl-acos.c = -fno-builtin-acosl
 CFLAGS-nldbl-acosh.c = -fno-builtin-acoshl
diff -urN a/sysdeps/sparc/sparc32/sparcv9/Makefile b/sysdeps/sparc/sparc32/sparcv9/Makefile
--- a/sysdeps/sparc/sparc32/sparcv9/Makefile	2014-04-26 21:47:08.578612000 +0100
+++ b/sysdeps/sparc/sparc32/sparcv9/Makefile	2014-04-26 21:47:34.868115600 +0100
@@ -10,11 +10,11 @@
 ASFLAGS-.os += -Wa,-Av9d
 ASFLAGS-.op += -Wa,-Av9d
 ASFLAGS-.og += -Wa,-Av9d
-ASFLAGS-.oS += -Wa,-Av9d
+ASFLAGS-.oSTATIC += -Wa,-Av9d
 else
 ASFLAGS-.o += -Wa,-Av9a
 ASFLAGS-.os += -Wa,-Av9a
 ASFLAGS-.op += -Wa,-Av9a
 ASFLAGS-.og += -Wa,-Av9a
-ASFLAGS-.oS += -Wa,-Av9a
+ASFLAGS-.oSTATIC += -Wa,-Av9a
 endif
diff -urN a/sysdeps/sparc/sparc64/Makefile b/sysdeps/sparc/sparc64/Makefile
--- a/sysdeps/sparc/sparc64/Makefile	2014-04-26 21:47:09.202647600 +0100
+++ b/sysdeps/sparc/sparc64/Makefile	2014-04-26 21:47:34.870115700 +0100
@@ -12,5 +12,5 @@
 ASFLAGS-.os += -Wa,-Av9d
 ASFLAGS-.op += -Wa,-Av9d
 ASFLAGS-.og += -Wa,-Av9d
-ASFLAGS-.oS += -Wa,-Av9d
+ASFLAGS-.oSTATIC += -Wa,-Av9d
 endif
