diff -pruN glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/bits/libc-vdso.h glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/bits/libc-vdso.h
--- glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/bits/libc-vdso.h	2012-12-25 08:32:13.000000000 +0530
+++ glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/bits/libc-vdso.h	2013-08-05 16:56:30.295860152 +0530
@@ -30,6 +30,8 @@ extern void *__vdso_clock_getres;
 
 extern void *__vdso_get_tbfreq;
 
+extern void *__vdso_getcpu;
+
 #endif
 
 #endif /* _LIBC_VDSO_H */
diff -pruN glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/init-first.c glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/init-first.c
--- glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/init-first.c	2012-12-25 08:32:13.000000000 +0530
+++ glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/init-first.c	2013-08-05 16:56:30.318860151 +0530
@@ -27,6 +27,7 @@ void *__vdso_gettimeofday attribute_hidd
 void *__vdso_clock_gettime;
 void *__vdso_clock_getres;
 void *__vdso_get_tbfreq;
+void *__vdso_getcpu;
 
 
 static inline void
@@ -41,6 +42,8 @@ _libc_vdso_platform_setup (void)
   __vdso_clock_getres = _dl_vdso_vsym ("__kernel_clock_getres", &linux2615);
 
   __vdso_get_tbfreq = _dl_vdso_vsym ("__kernel_vdso_get_tbfreq", &linux2615);
+
+  __vdso_getcpu = _dl_vdso_vsym ("__kernel_getcpu", &linux2615);
 }
 
 # define VDSO_SETUP _libc_vdso_platform_setup
diff -pruN glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/sched_getcpu.c glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/sched_getcpu.c
--- glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/sched_getcpu.c	1970-01-01 05:30:00.000000000 +0530
+++ glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/sched_getcpu.c	2013-08-05 16:56:30.319860151 +0530
@@ -0,0 +1,30 @@
+/* Copyright (C) 2013 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <sched.h>
+#include <sysdep.h>
+#include <bits/libc-vdso.h>
+
+
+int
+sched_getcpu (void)
+{
+  unsigned int cpu;
+  int r = INLINE_VSYSCALL (getcpu, 3, &cpu, NULL, NULL);
+
+  return r == -1 ? r : cpu;
+}
diff -pruN glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/Versions glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/Versions
--- glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/Versions	2012-12-25 08:32:13.000000000 +0530
+++ glibc-2.17-c758a686/sysdeps/unix/sysv/linux/powerpc/Versions	2013-08-05 16:56:30.294860152 +0530
@@ -3,5 +3,6 @@ libc {
     __vdso_get_tbfreq;
     __vdso_clock_gettime;
     __vdso_clock_getres;
+    __vdso_getcpu;
   }
 }
