diff -urN a/lib/Support/Errno.cpp b/lib/Support/Errno.cpp
--- a/lib/Support/Errno.cpp	2011-02-09 04:18:48.000000000 +0000
+++ b/lib/Support/Errno.cpp	2013-09-09 00:51:04.442462700 +0100
@@ -52,7 +52,7 @@
 # endif
 #elif HAVE_DECL_STRERROR_S // "Windows Secure API"
     if (errnum)
-      strerror_s(buffer, errnum);
+      strerror_s(buffer, MaxErrStrLen-1, errnum);
 #elif defined(HAVE_STRERROR)
   // Copy the thread un-safe result of strerror into
   // the buffer as fast as possible to minimize impact
diff -urN a/lib/Support/Windows/Windows.h b/lib/Support/Windows/Windows.h
--- a/lib/Support/Windows/Windows.h	2011-02-09 04:18:30.000000000 +0000
+++ b/lib/Support/Windows/Windows.h	2013-09-09 00:52:43.340521100 +0100
@@ -20,7 +20,13 @@
 #undef _WIN32_WINNT
 
 // Require at least Windows 2000 API.
-#define _WIN32_WINNT 0x0500
+#if !defined(_WIN32_WINNT)
+ #define _WIN32_WINNT 0x0500
+#else
+ #if (_WIN32_WINNT < 0x0500)
+  #error Require at least Windows 2000 API.
+ #endif
+#endif
 #define _WIN32_IE    0x0500 // MinGW at it again.
 #define WIN32_LEAN_AND_MEAN
 
