diff -urN a/lib/Driver/ToolChain.cpp b/lib/Driver/ToolChain.cpp
--- a/lib/Driver/ToolChain.cpp	2011-11-07 11:15:01.000000000 +0100
+++ b/lib/Driver/ToolChain.cpp	2013-09-07 11:29:23.820677474 +0200
@@ -38,6 +38,17 @@
 }
 
 std::string ToolChain::GetProgramPath(const char *Name, bool WantFile) const {
+  // Don't try to prefix clang
+  if(strcmp(Name, "clang")) {
+    // Try first with prefixed by triple
+    std::string targetName = Host.getTriple().getTriple();
+    targetName += "-";
+    targetName += Name;
+    std::string targetProgram = Host.getDriver().GetProgramPath(targetName.c_str(), *this, WantFile);
+    if(targetProgram != targetName) {
+      return targetProgram;
+    }
+  }
   return Host.getDriver().GetProgramPath(Name, *this, WantFile);
 }
 
