diff -urN b/include/clang/Driver/ToolChain.h c/include/clang/Driver/ToolChain.h
--- b/include/clang/Driver/ToolChain.h	2013-09-13 13:07:41.861314642 +0200
+++ c/include/clang/Driver/ToolChain.h	2013-09-13 13:08:28.148291982 +0200
@@ -74,6 +74,7 @@
   }
 
   virtual void AddLinkSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const;
+  virtual void AddCPlusPlusIncludeSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const;
   virtual void AddIncludeSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const;
   
   path_list &getFilePaths() { return FilePaths; }
diff -urN b/lib/Driver/ToolChain.cpp c/lib/Driver/ToolChain.cpp
--- b/lib/Driver/ToolChain.cpp	2013-09-13 13:07:41.861314642 +0200
+++ c/lib/Driver/ToolChain.cpp	2013-09-13 13:08:28.148291982 +0200
@@ -57,6 +57,23 @@
   CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include"));
 }
 
+void ToolChain::AddCPlusPlusIncludeSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const {
+  llvm::sys::Path P(getDriver().Dir);
+  std::string GccVersion(CLANG_GCC_VERSION_STR);
+
+  // Checks
+  if (P.isEmpty() || GccVersion.empty()) {
+    return;
+  }
+
+  P.eraseComponent(); // Remove /bin from foo/bin
+  std::string BaseDir = P.str() + std::string("/lib/gcc/") + Host.getTriple().getTriple() + std::string("/") + GccVersion;
+
+  CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include/c++/" + GccVersion));
+  CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include/c++/" + GccVersion + "/" + Host.getTriple().getTriple()));
+  CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include/c++/" + GccVersion + "/backward"));
+}
+
 void ToolChain::AddLinkSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const {
   llvm::sys::Path P(getDriver().Dir);
   std::string GccVersion(CLANG_GCC_VERSION_STR);
diff -urN b/lib/Driver/Tools.cpp c/lib/Driver/Tools.cpp
--- b/lib/Driver/Tools.cpp	2013-09-13 13:07:41.865314554 +0200
+++ c/lib/Driver/Tools.cpp	2013-09-13 13:08:28.148291982 +0200
@@ -1543,6 +1543,9 @@
   if (isa<LinkJobAction>(JA)) {
      getToolChain().AddLinkSearchPathArgs(Args, CmdArgs);
   } else {
+     if (types::isCXX(InputType)) {
+       getToolChain().AddCPlusPlusIncludeSearchPathArgs(Args, CmdArgs);
+     }
      getToolChain().AddIncludeSearchPathArgs(Args, CmdArgs);
   }
 
diff -urN b/lib/Frontend/InitHeaderSearch.cpp c/lib/Frontend/InitHeaderSearch.cpp
--- b/lib/Frontend/InitHeaderSearch.cpp	2011-11-07 11:27:39.000000000 +0100
+++ c/lib/Frontend/InitHeaderSearch.cpp	2013-09-13 13:09:51.090471819 +0200
@@ -303,17 +303,13 @@
   case llvm::Triple::MinGW32: { 
       // mingw-w64 crt include paths
       llvm::sys::Path P(HSOpts.ResourceDir);
-      P.appendComponent("../../../i686-w64-mingw32/include"); // <sysroot>/i686-w64-mingw32/include
-      AddPath(P.str(), System, true, false, false);
-      P = llvm::sys::Path(HSOpts.ResourceDir);
-      P.appendComponent("../../../x86_64-w64-mingw32/include"); // <sysroot>/x86_64-w64-mingw32/include
+      P.appendComponent("../../../" + triple.getTriple() + "/include"); // <sysroot>/i686-w64-mingw32/include
       AddPath(P.str(), System, true, false, false);
       // mingw.org crt include paths
       P = llvm::sys::Path(HSOpts.ResourceDir);
       P.appendComponent("../../../include"); // <sysroot>/include
       AddPath(P.str(), System, true, false, false);
       AddPath("/mingw/include", System, true, false, false);
-      AddPath("c:/mingw/include", System, true, false, false); 
     }
     break;
       
