diff -urN gcc-5666.3/gcc/config/arm/arm.c gcc-5666.3.mod2/gcc/config/arm/arm.c
--- gcc-5666.3/gcc/config/arm/arm.c	2010-10-14 22:27:52.000000000 +0200
+++ gcc-5666.3.mod2/gcc/config/arm/arm.c	2012-11-06 11:52:56.415784889 +0100
@@ -24,7 +24,7 @@
 
 /* APPLE LOCAL begin 6902792 Q register clobbers in inline asm */
 #include <stdlib.h>
-#include <ctype.h>
+//#include <ctype.h>
 /* APPLE LOCAL end 6902792 Q register clobbers in inline asm */
 #include "config.h"
 #include "system.h"
