diff -urN a/gcc/configure b/gcc/configure
--- a/gcc/configure	2013-05-31 00:54:28.145299683 +0100
+++ b/gcc/configure	2013-05-31 00:54:28.418301835 +0100
@@ -9702,6 +9702,37 @@
 _ACEOF
 
 
+fi
+
+# Try to determine the array type of the second argument of getgroups
+# for the target system (int or gid_t).
+echo "$as_me:$LINENO: checking for gid_t in sys/types.h" >&5
+echo $ECHO_N "checking for gid_t in sys/types.h... $ECHO_C" >&6
+if test "${ac_cv_type_gid_t+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+#include <sys/types.h>
+
+_ACEOF
+if (eval "$ac_cpp conftest.$ac_ext") 2>&5 |
+  $EGREP "gid_t" >/dev/null 2>&1; then
+  ac_cv_type_gid_t=yes
+else
+  ac_cv_type_gid_t=no
+fi
+rm -f conftest*
+
+fi
+echo "$as_me:$LINENO: result: $ac_cv_type_gid_t" >&5
+echo "${ECHO_T}$ac_cv_type_gid_t" >&6
+if test $ac_cv_type_gid_t = no; then
+
 cat >>confdefs.h <<\_ACEOF
 #define gid_t int
 _ACEOF
@@ -11563,6 +11594,7 @@
 #define $ac_tr_decl 1
 
 #include "ansidecl.h"
+#include "unistd.h"
 #include "system.h"
 
 int
@@ -14344,7 +14376,7 @@
 echo "${ECHO_T}$gcc_cv_as_eh_frame" >&6
 
 
-if test $gcc_cv_as_eh_frame = buggy; then
+if test $gcc_cv_as_eh_frame = buggy && test x$host = x$target ; then
 
 cat >>confdefs.h <<\_ACEOF
 #define USE_AS_TRADITIONAL_FORMAT 1
@@ -16238,6 +16270,9 @@
         fi
         ;;
        *-*-darwin*)
+      if test x$host != x$target || test "x$TARGET_SYSTEM_ROOT" != x; then
+          gcc_cv_libc_provides_ssp=yes
+      else
          echo "$as_me:$LINENO: checking for __stack_chk_fail" >&5
 echo $ECHO_N "checking for __stack_chk_fail... $ECHO_C" >&6
 if test "${ac_cv_func___stack_chk_fail+set}" = set; then
@@ -16333,6 +16368,7 @@
 else
   echo "no __stack_chk_fail on this target"
 fi
+fi
 
         ;;
   *) gcc_cv_libc_provides_ssp=no ;;
diff -urN a/gcc/configure.ac b/gcc/configure.ac
--- a/gcc/configure.ac	2013-05-31 00:54:28.140299644 +0100
+++ b/gcc/configure.ac	2013-05-31 00:54:28.412301788 +0100
@@ -1130,6 +1130,7 @@
 
 gcc_AC_CHECK_DECLS(getrlimit setrlimit getrusage, , ,[
 #include "ansidecl.h"
+#include "unistd.h"
 #include "system.h"
 #ifdef HAVE_SYS_RESOURCE_H
 #include <sys/resource.h>
@@ -2256,7 +2257,7 @@
     gcc_cv_as_eh_frame=no
   fi])
 
-if test $gcc_cv_as_eh_frame = buggy; then
+if test $gcc_cv_as_eh_frame = buggy && test x$host = x$target ; then
   AC_DEFINE(USE_AS_TRADITIONAL_FORMAT, 1,
   [Define if your assembler mis-optimizes .eh_frame data.])
 fi
@@ -3252,8 +3253,12 @@
         fi
         ;;
        *-*-darwin*)
+      if test x$host != x$target || test "x$TARGET_SYSTEM_ROOT" != x; then
+          gcc_cv_libc_provides_ssp=yes
+      else
          AC_CHECK_FUNC(__stack_chk_fail,[gcc_cv_libc_provides_ssp=yes],
            [echo "no __stack_chk_fail on this target"])
+      fi
         ;;
   *) gcc_cv_libc_provides_ssp=no ;;
     esac])
