diff -urN a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
--- a/lib/Driver/ToolChains.cpp	2012-05-12 02:16:02.000000000 +0200
+++ b/lib/Driver/ToolChains.cpp	2013-09-09 21:53:55.402444645 +0200
@@ -163,6 +163,12 @@
 
     return "arm";
   }
+  case llvm::Triple::x86: 
+    return "i386";
+  case llvm::Triple::ppc:
+    return "ppc";
+  case llvm::Triple::ppc64:
+    return "ppc64";
   }
 }
 
diff -urN a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
--- a/lib/Driver/Tools.cpp	2012-04-18 23:32:25.000000000 +0200
+++ b/lib/Driver/Tools.cpp	2013-09-09 21:58:28.529113005 +0200
@@ -2872,17 +2872,6 @@
 
   // If using a driver driver, force the arch.
   const std::string &Arch = getToolChain().getArchName();
-  if (getToolChain().getTriple().isOSDarwin()) {
-    CmdArgs.push_back("-arch");
-
-    // FIXME: Remove these special cases.
-    if (Arch == "powerpc")
-      CmdArgs.push_back("ppc");
-    else if (Arch == "powerpc64")
-      CmdArgs.push_back("ppc64");
-    else
-      CmdArgs.push_back(Args.MakeArgString(Arch));
-  }
 
   // Try to force gcc to match the tool chain we want, if we recognize
   // the arch.
