diff -urN a/runtime/compiler-rt/Makefile b/runtime/compiler-rt/Makefile
--- a/runtime/compiler-rt/Makefile	2012-03-05 21:19:03.000000000 +0100
+++ b/runtime/compiler-rt/Makefile	2013-11-08 17:11:44.711159667 +0100
@@ -61,7 +61,7 @@
 #     information that compiler-rt would need to build libraries for all the
 #     requested targets. This might be a simple short term solution, but is
 #     likely to be unwieldly and irritating to maintain in the long term.
-ifneq ($(LLVM_CROSS_COMPILING),1)
+#ifneq ($(LLVM_CROSS_COMPILING),1)
 ifneq ($(CLANG_NO_RUNTIME),1)
 ifeq ($(shell test -d $(COMPILERRT_SRC_ROOT) && echo OK),OK)
 
@@ -71,7 +71,7 @@
 # build/install runtime libraries for as many targets as clang was configured to
 # support.
 RuntimeDirs :=
-ifeq ($(OS),Darwin)
+ifeq ($(TARGET_OS),Darwin)
 RuntimeDirs += darwin
 RuntimeLibrary.darwin.Configs := \
 	eprintf 10.4 osx ios cc_kext \
@@ -79,16 +79,16 @@
 endif
 
 # On Linux, include a library which has all the runtime functions.
-ifeq ($(OS),Linux)
+ifeq ($(TARGET_OS),Linux)
 RuntimeDirs += linux
 RuntimeLibrary.linux.Configs :=
 
 # We currently only try to generate runtime libraries on x86.
-ifeq ($(ARCH),x86)
+ifeq ($(TARGET_ARCH),x86)
 RuntimeLibrary.linux.Configs += \
 	full-i386 profile-i386 asan-i386
 endif
-ifeq ($(ARCH),x86_64)
+ifeq ($(TARGET_ARCH),x86_64)
 RuntimeLibrary.linux.Configs += \
 	full-x86_64 profile-x86_64 asan-x86_64
 endif
@@ -160,4 +160,4 @@
 
 endif
 endif
-endif
+#endif
