diff -urN a/lib/Support/Unix/Host.inc b/lib/Support/Unix/Host.inc
--- a/lib/Support/Unix/Host.inc	2012-01-05 19:28:46.000000000 +0100
+++ b/lib/Support/Unix/Host.inc	2013-09-10 13:01:41.651537956 +0200
@@ -46,18 +46,5 @@
   Triple += '-';
   Triple += ArchSplit.second;
 
-  // Force i<N>86 to i386.
-  if (Triple[0] == 'i' && isdigit(Triple[1]) &&
-      Triple[2] == '8' && Triple[3] == '6')
-    Triple[1] = '3';
-
-  // On darwin, we want to update the version to match that of the
-  // target.
-  std::string::size_type DarwinDashIdx = Triple.find("-darwin");
-  if (DarwinDashIdx != std::string::npos) {
-    Triple.resize(DarwinDashIdx + strlen("-darwin"));
-    Triple += getOSVersion();
-  }
-
   return Triple;
 }
