diff -urN a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
--- a/lib/Driver/ToolChains.cpp	2013-05-06 18:26:41.000000000 +0200
+++ b/lib/Driver/ToolChains.cpp	2013-09-11 09:09:03.599690889 +0200
@@ -147,6 +147,12 @@
 
     return "arm";
   }
+  case llvm::Triple::x86: 
+    return "i386";
+  case llvm::Triple::ppc:
+    return "ppc";
+  case llvm::Triple::ppc64:
+    return "ppc64";
   }
 }
 
diff -urN a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
--- a/lib/Driver/Tools.cpp	2013-05-30 20:01:30.000000000 +0200
+++ b/lib/Driver/Tools.cpp	2013-09-11 09:09:03.599690889 +0200
@@ -3807,17 +3807,6 @@
 
   // If using a driver driver, force the arch.
   llvm::Triple::ArchType Arch = getToolChain().getArch();
-  if (getToolChain().getTriple().isOSDarwin()) {
-    CmdArgs.push_back("-arch");
-
-    // FIXME: Remove these special cases.
-    if (Arch == llvm::Triple::ppc)
-      CmdArgs.push_back("ppc");
-    else if (Arch == llvm::Triple::ppc64)
-      CmdArgs.push_back("ppc64");
-    else
-      CmdArgs.push_back(Args.MakeArgString(getToolChain().getArchName()));
-  }
 
   // Try to force gcc to match the tool chain we want, if we recognize
   // the arch.
