diff -urN a/include/clang/Analysis/Support/BumpVector.h b/include/clang/Analysis/Support/BumpVector.h
--- a/include/clang/Analysis/Support/BumpVector.h	2010-02-21 23:14:52.000000000 +0100
+++ b/include/clang/Analysis/Support/BumpVector.h	2013-09-02 20:27:23.248577650 +0200
@@ -24,6 +24,7 @@
 #include "llvm/ADT/PointerIntPair.h"
 #include <algorithm>
 #include <cstring>
+#include <memory>
 
 namespace clang {
   
