diff -Nru glibc-2.17-c758a686/nscd/nscd.conf glibc-2.17-c758a686/nscd/nscd.conf
--- glibc-2.17-c758a686/nscd/nscd.conf	2012-06-05 07:42:49.000000000 -0600
+++ glibc-2.17-c758a686/nscd/nscd.conf	2012-06-07 12:15:21.818318670 -0600
@@ -33,7 +33,7 @@
 #	logfile			/var/log/nscd.log
 #	threads			4
 #	max-threads		32
-#	server-user		nobody
+	server-user		nscd
 #	stat-user		somebody
 	debug-level		0
 #	reload-count		5
