--- llvmCore/include/llvm/ADT/SmallVector.h.orig	2012-03-24 15:19:42 +0000
+++ llvmCore/include/llvm/ADT/SmallVector.h	2012-03-24 15:19:47 +0000
@@ -18,6 +18,7 @@
 #include <algorithm>
 #include <cassert>
 #include <cstring>
+#include <cstddef>
 #include <memory>
 
 #ifdef _MSC_VER
--- llvmCore/include/llvm/Use.h.orig	2012-05-08 20:37:45.575483212 +0100
+++ llvmCore/include/llvm/Use.h	2012-05-08 20:38:39.408817144 +0100
@@ -25,6 +25,7 @@
 #ifndef LLVM_USE_H
 #define LLVM_USE_H
 
+#include <cstddef>
 #include "llvm/Support/Casting.h"
 #include "llvm/ADT/PointerIntPair.h"
 #include <iterator>
--- llvmCore/include/llvm/ADT/DenseMap.h.orig	2012-05-08 20:41:51.912011653 +0100
+++ llvmCore/include/llvm/ADT/DenseMap.h	2012-05-08 20:42:11.536133539 +0100
@@ -14,6 +14,7 @@
 #ifndef LLVM_ADT_DENSEMAP_H
 #define LLVM_ADT_DENSEMAP_H
 
+#include <cstddef>
 #include "llvm/Support/MathExtras.h"
 #include "llvm/Support/PointerLikeTypeTraits.h"
 #include "llvm/Support/type_traits.h"
--- llvmCore/include/llvm/ADT/ilist.h.orig	2012-05-08 20:46:54.867184093 +0100
+++ llvmCore/include/llvm/ADT/ilist.h	2012-05-08 20:48:02.391828322 +0100
@@ -38,6 +38,7 @@
 #ifndef LLVM_ADT_ILIST_H
 #define LLVM_ADT_ILIST_H
 
+#include <cstddef>
 #include <cassert>
 #include <iterator>
 
--- llvmCore/lib/ExecutionEngine/JIT/Intercept.cpp.orig	2012-05-08 21:36:55.818349473 +0100
+++ llvmCore/lib/ExecutionEngine/JIT/Intercept.cpp	2012-05-08 21:37:11.670462092 +0100
@@ -15,6 +15,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <unistd.h>
 #include "JIT.h"
 #include "llvm/Support/ErrorHandling.h"
 #include "llvm/System/DynamicLibrary.h"
