diff -urN a/configure b/configure
--- a/configure	2014-02-23 14:14:09.194633800 +0000
+++ b/configure	2014-02-23 14:14:19.480939900 +0000
@@ -4386,6 +4386,16 @@
 $as_echo "no, using $LN_S" >&6; }
 fi
 
+# Even though 'ln -s' doesn't work right in MSYS2, it works enough for
+# eglibc, whereas cp -r doesn't.
+case "$build_os" in
+*mingw*)
+  as_ln_s='ln -s'
+  LN_S=$as_ln_s
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Hacked as_ln_s and LN_S to 'ln -s' for MSYS2" >&5
+$as_echo "$as_me: WARNING: Hacked as_ln_s and LN_S to 'ln -s' for MSYS2" >&2;}
+  ;;
+esac
 
 # Was a --with-binutils option given?
 if test -n "$path_binutils"; then
diff -urN a/configure.in b/configure.in
--- a/configure.in	2014-02-23 14:14:09.195133800 +0000
+++ b/configure.in	2014-02-23 14:14:19.484440400 +0000
@@ -938,6 +938,15 @@
   INSTALL='\$(..)./scripts/install-sh -c'
 fi
 AC_PROG_LN_S
+# Even though 'ln -s' doesn't work right in MSYS2, it works enough for
+# eglibc, whereas cp -r doesn't.
+case "$build_os" in
+*mingw*)
+  as_ln_s='ln -s'
+  LN_S=$as_ln_s
+  AC_MSG_WARN(Hacked as_ln_s and LN_S to 'ln -s' for MSYS2)
+  ;;
+esac
 
 LIBC_PROG_BINUTILS
 AC_SUBST(MIG)dnl Needed by sysdeps/mach/configure.in
