diff -urN a/NEWS b/NEWS
--- a/NEWS	2013-08-10 23:52:55.000000000 +0100
+++ b/NEWS	2014-02-23 14:21:47.567839800 +0000
@@ -5,6 +5,12 @@
 Please send GNU C library bug reports via <http://sourceware.org/bugzilla/>
 using `glibc' in the "product" field.
 
+Version 2.18.1
+
+* The following bugs are resolved with this release:
+
+  15909.
+
 Version 2.18
 
 * The following bugs are resolved with this release:
diff -urN a/ports/ChangeLog.arm b/ports/ChangeLog.arm
--- a/ports/ChangeLog.arm	2013-08-10 23:52:55.000000000 +0100
+++ b/ports/ChangeLog.arm	2014-02-23 14:21:47.569339900 +0000
@@ -1,3 +1,9 @@
+2013-08-30  Will Newton  <will.newton@linaro.org>
+
+	[BZ #15909]
+	* sysdeps/unix/sysv/linux/arm/clone.S (__clone): Use blx
+	instead of mov lr, pc.
+
 2013-07-03  Joseph Myers  <joseph@codesourcery.com>
 
 	* sysdeps/arm/include/bits/setjmp.h [_ISOMAC] (JMP_BUF_REGLIST):
diff -urN a/ports/sysdeps/unix/sysv/linux/arm/clone.S b/ports/sysdeps/unix/sysv/linux/arm/clone.S
--- a/ports/sysdeps/unix/sysv/linux/arm/clone.S	2013-08-10 23:52:55.000000000 +0100
+++ b/ports/sysdeps/unix/sysv/linux/arm/clone.S	2014-02-23 14:21:47.570340100 +0000
@@ -93,8 +93,8 @@
 	mov	lr, pc
 	bx      ip
 #else
-	mov	lr, pc
-	ldr 	pc, [sp], #8
+	ldr 	lr, [sp], #8
+	blx	lr
 #endif
 
 	@ and we are done, passing the return value through r0
