diff -urN a/NEWS b/NEWS
--- a/NEWS	2014-02-23 14:21:54.507721000 +0000
+++ b/NEWS	2014-02-23 14:22:06.075189900 +0000
@@ -9,7 +9,7 @@
 
 * The following bugs are resolved with this release:
 
-  15909.
+  15909, 15996.
 
 Version 2.18
 
diff -urN a/nptl/ChangeLog b/nptl/ChangeLog
--- a/nptl/ChangeLog	2014-02-23 14:21:54.545225800 +0000
+++ b/nptl/ChangeLog	2014-02-23 14:22:06.080190500 +0000
@@ -1,3 +1,12 @@
+2013-10-03  Siddhesh Poyarekar  <siddhesh@redhat.com>
+
+	[BZ #15996]
+	* pthread_cond_broadcast.c (__pthread_cond_broadcast)
+	[lll_futex_cmp_requeue_pi && __ASSUME_REQUEUE_PI]: Use
+	USE_REQUEUE_PI.
+	* pthread_cond_signal.c (__pthread_cond_signal)
+	[lll_futex_cmd_requeue_pi && __ASSUME_REQUEUE_PI]: Likewise.
+
 2013-07-23  David S. Miller  <davem@davemloft.net>
 
 	* tst-cancel4.c (WRITE_BUFFER_SIZE): Adjust comment.
diff -urN a/nptl/pthread_cond_broadcast.c b/nptl/pthread_cond_broadcast.c
--- a/nptl/pthread_cond_broadcast.c	2014-02-23 14:21:54.572229200 +0000
+++ b/nptl/pthread_cond_broadcast.c	2014-02-23 14:22:06.081190700 +0000
@@ -63,10 +63,7 @@
 
 #if (defined lll_futex_cmp_requeue_pi \
      && defined __ASSUME_REQUEUE_PI)
-      int pi_flag = PTHREAD_MUTEX_PRIO_INHERIT_NP | PTHREAD_MUTEX_ROBUST_NP;
-      pi_flag &= mut->__data.__kind;
-
-      if (pi_flag == PTHREAD_MUTEX_PRIO_INHERIT_NP)
+      if (USE_REQUEUE_PI (mut))
 	{
 	  if (lll_futex_cmp_requeue_pi (&cond->__data.__futex, 1, INT_MAX,
 					&mut->__data.__lock, futex_val,
diff -urN a/nptl/pthread_cond_signal.c b/nptl/pthread_cond_signal.c
--- a/nptl/pthread_cond_signal.c	2014-02-23 14:21:54.573229300 +0000
+++ b/nptl/pthread_cond_signal.c	2014-02-23 14:22:06.082190800 +0000
@@ -49,14 +49,9 @@
 
 #if (defined lll_futex_cmp_requeue_pi \
      && defined __ASSUME_REQUEUE_PI)
-      int pi_flag = PTHREAD_MUTEX_PRIO_INHERIT_NP | PTHREAD_MUTEX_ROBUST_NP;
       pthread_mutex_t *mut = cond->__data.__mutex;
 
-      /* Do not use requeue for pshared condvars.  */
-      if (mut != (void *) ~0l)
-	pi_flag &= mut->__data.__kind;
-
-      if (__builtin_expect (pi_flag == PTHREAD_MUTEX_PRIO_INHERIT_NP, 0)
+      if (USE_REQUEUE_PI (mut)
 	/* This can only really fail with a ENOSYS, since nobody can modify
 	   futex while we have the cond_lock.  */
 	  && lll_futex_cmp_requeue_pi (&cond->__data.__futex, 1, 0,
