diff -urN a/make/platform/clang_darwin.mk b/make/platform/clang_darwin.mk
--- a/make/platform/clang_darwin.mk	2012-10-24 16:03:41.000000000 +0200
+++ b/make/platform/clang_darwin.mk	2013-09-09 11:47:03.303889499 +0200
@@ -80,12 +80,14 @@
 # object files. If we are on that platform, strip out all ARM archs. We still
 # build the libraries themselves so that Clang can find them where it expects
 # them, even though they might not have an expected slice.
+ifneq ($(shell which sw_vers),)
 ifneq ($(shell sw_vers -productVersion | grep 10.6),)
 UniversalArchs.ios := $(filter-out armv7, $(UniversalArchs.ios))
 UniversalArchs.cc_kext := $(filter-out armv7, $(UniversalArchs.cc_kext))
 UniversalArchs.cc_kext_ios5 := $(filter-out armv7, $(UniversalArchs.cc_kext_ios5))
 UniversalArchs.profile_ios := $(filter-out armv7, $(UniversalArchs.profile_ios))
 endif
+endif
 
 # If RC_SUPPORTED_ARCHS is defined, treat it as a list of the architectures we
 # are intended to support and limit what we try to build to that.
