diff -urN c/lib/Driver/ToolChain.cpp d/lib/Driver/ToolChain.cpp
--- c/lib/Driver/ToolChain.cpp	2010-08-24 00:35:37.000000000 +0200
+++ d/lib/Driver/ToolChain.cpp	2013-09-04 22:39:22.813724681 +0200
@@ -36,6 +36,17 @@
 }
 
 std::string ToolChain::GetProgramPath(const char *Name, bool WantFile) const {
+  // Don't try to prefix clang
+  if(strcmp(Name, "clang")) {
+    // Try first with prefixed by triple
+    std::string targetName = Host.getTriple().getTriple();
+    targetName += "-";
+    targetName += Name;
+    std::string targetProgram = Host.getDriver().GetProgramPath(targetName.c_str(), *this, WantFile);
+    if(targetProgram != targetName) {
+      return targetProgram;
+    }
+  }
   return Host.getDriver().GetProgramPath(Name, *this, WantFile);
 }
 
