diff -urN b/include/clang/Driver/ToolChain.h c/include/clang/Driver/ToolChain.h
--- b/include/clang/Driver/ToolChain.h	2013-09-13 10:23:15.077654046 +0200
+++ c/include/clang/Driver/ToolChain.h	2013-09-13 10:23:27.681244043 +0200
@@ -73,6 +73,7 @@
   }
 
   virtual void AddLinkSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const;
+  virtual void AddCPlusPlusIncludeSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const;
   virtual void AddIncludeSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const;
   
   path_list &getFilePaths() { return FilePaths; }
diff -urN b/lib/Driver/ToolChain.cpp c/lib/Driver/ToolChain.cpp
--- b/lib/Driver/ToolChain.cpp	2013-09-13 10:23:15.077654046 +0200
+++ c/lib/Driver/ToolChain.cpp	2013-09-13 10:23:27.685243914 +0200
@@ -56,6 +56,23 @@
   CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include"));
 }
 
+void ToolChain::AddCPlusPlusIncludeSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const {
+  llvm::sys::Path P(getDriver().Dir);
+  std::string GccVersion(CLANG_GCC_VERSION_STR);
+
+  // Checks
+  if (P.isEmpty() || GccVersion.empty()) {
+    return;
+  }
+
+  P.eraseComponent(); // Remove /bin from foo/bin
+  std::string BaseDir = P.str() + std::string("/lib/gcc/") + Host.getTriple().getTriple() + std::string("/") + GccVersion;
+
+  CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include/c++/" + GccVersion));
+  CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include/c++/" + GccVersion + "/" + Host.getTriple().getTriple()));
+  CmdArgs.push_back(Args.MakeArgString("-I" + BaseDir + "/../../../../" + Host.getTriple().getTriple() + "/include/c++/" + GccVersion + "/backward"));
+}
+
 void ToolChain::AddLinkSearchPathArgs(const ArgList &Args, ArgStringList &CmdArgs) const {
   llvm::sys::Path P(getDriver().Dir);
   std::string GccVersion(CLANG_GCC_VERSION_STR);
diff -urN b/lib/Driver/Tools.cpp c/lib/Driver/Tools.cpp
--- b/lib/Driver/Tools.cpp	2013-09-13 10:23:15.077654046 +0200
+++ c/lib/Driver/Tools.cpp	2013-09-13 10:23:27.689243785 +0200
@@ -1268,6 +1268,9 @@
   if (isa<LinkJobAction>(JA)) {
      getToolChain().AddLinkSearchPathArgs(Args, CmdArgs);
   } else {
+     if (types::isCXX(InputType)) {
+       getToolChain().AddCPlusPlusIncludeSearchPathArgs(Args, CmdArgs);
+     }
      getToolChain().AddIncludeSearchPathArgs(Args, CmdArgs);
   }
 
diff -urN b/lib/Frontend/InitHeaderSearch.cpp c/lib/Frontend/InitHeaderSearch.cpp
--- b/lib/Frontend/InitHeaderSearch.cpp	2011-03-22 01:24:04.000000000 +0100
+++ c/lib/Frontend/InitHeaderSearch.cpp	2013-09-13 10:48:57.813058453 +0200
@@ -549,10 +549,9 @@
   case llvm::Triple::MinGW32:
     // FIXME: We should be aware of i686-w64-mingw32.
     if (triple.getArch() == llvm::Triple::x86_64)
-      AddPath("c:/mingw/x86_64-w64-mingw32/include",
+      AddPath("/mingw/" + triple.getTriple() + "/include",
               System, true, false, false);
     AddPath("/mingw/include", System, true, false, false);
-    AddPath("c:/mingw/include", System, true, false, false);
     break;
   default:
     break;
