diff -urN a/gcc/config/i386/xm-mingw32.h b/gcc/config/i386/xm-mingw32.h
--- a/gcc/config/i386/xm-mingw32.h	2013-05-31 00:32:24.425870093 +0100
+++ b/gcc/config/i386/xm-mingw32.h	2013-05-31 00:32:25.068875192 +0100
@@ -33,4 +33,8 @@
 
 /* MSVCRT does not support the "ll" format specifier for printing
    "long long" values.  Instead, we use "I64".  */
-#define HOST_LONG_LONG_FORMAT "I64"
+#ifndef __USE_MINGW_ANSI_STDIO
+ #define HOST_LONG_LONG_FORMAT "I64"
+#else
+ #define HOST_LONG_LONG_FORMAT "ll"
+#endif
