diff -urN a/gcc/config/i386/t-darwin b/gcc/config/i386/t-darwin
--- a/gcc/config/i386/t-darwin	2013-06-19 00:06:23.158418615 +0200
+++ b/gcc/config/i386/t-darwin	2013-06-26 18:50:48.635031833 +0200
@@ -3,12 +3,10 @@
 LIB1ASMSRC = i386/lib1funcs.asm
 LIB1ASMFUNCS = _get_pc_thunk_ax _get_pc_thunk_dx _get_pc_thunk_cx _get_pc_thunk_bx _get_pc_thunk_si _get_pc_thunk_di _get_pc_thunk_bp
 # APPLE LOCAL end 4099000
-# APPLE LOCAL avoid try fat on thin system
-ifneq ($(shell lipo -info $(prefix)/lib/libSystem.B.dylib | grep x86_64),)
+
 MULTILIB_OPTIONS = m64
 MULTILIB_DIRNAMES = x86_64
-# APPLE LOCAL avoid try fat on thin system
-endif
+
 LIB2_SIDITI_CONV_FUNCS=yes
 LIB2FUNCS_EXTRA = $(srcdir)/config/darwin-64.c
 
