diff -urN a/include/llvm/ADT/DenseMap.h b/include/llvm/ADT/DenseMap.h
--- a/include/llvm/ADT/DenseMap.h	2010-09-01 16:00:35.000000000 +0200
+++ b/include/llvm/ADT/DenseMap.h	2013-09-02 20:16:42.295245559 +0200
@@ -14,6 +14,7 @@
 #ifndef LLVM_ADT_DENSEMAP_H
 #define LLVM_ADT_DENSEMAP_H
 
+#include <cstddef>
 #include "llvm/Support/MathExtras.h"
 #include "llvm/Support/PointerLikeTypeTraits.h"
 #include "llvm/Support/type_traits.h"
diff -urN a/include/llvm/ADT/ilist.h b/include/llvm/ADT/ilist.h
--- a/include/llvm/ADT/ilist.h	2010-08-29 16:48:15.000000000 +0200
+++ b/include/llvm/ADT/ilist.h	2013-09-02 20:16:42.295245559 +0200
@@ -38,6 +38,7 @@
 #ifndef LLVM_ADT_ILIST_H
 #define LLVM_ADT_ILIST_H
 
+#include <cstddef>
 #include <cassert>
 #include <cstddef>
 #include <iterator>
diff -urN a/include/llvm/ADT/SmallVector.h b/include/llvm/ADT/SmallVector.h
--- a/include/llvm/ADT/SmallVector.h	2010-09-30 20:05:47.000000000 +0200
+++ b/include/llvm/ADT/SmallVector.h	2013-09-02 20:16:42.295245559 +0200
@@ -20,6 +20,7 @@
 #include <cstddef>
 #include <cstdlib>
 #include <cstring>
+#include <cstddef>
 #include <memory>
 
 #ifdef _MSC_VER
diff -urN a/include/llvm/Use.h b/include/llvm/Use.h
--- a/include/llvm/Use.h	2010-07-26 20:48:07.000000000 +0200
+++ b/include/llvm/Use.h	2013-09-02 20:16:42.295245559 +0200
@@ -25,6 +25,7 @@
 #ifndef LLVM_USE_H
 #define LLVM_USE_H
 
+#include <cstddef>
 #include "llvm/Support/Casting.h"
 #include "llvm/ADT/PointerIntPair.h"
 #include <cstddef>
diff -urN a/lib/ExecutionEngine/JIT/Intercept.cpp b/lib/ExecutionEngine/JIT/Intercept.cpp
--- a/lib/ExecutionEngine/JIT/Intercept.cpp	2010-08-30 23:54:03.000000000 +0200
+++ b/lib/ExecutionEngine/JIT/Intercept.cpp	2013-09-02 20:16:42.299245758 +0200
@@ -15,6 +15,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <unistd.h>
 #include "JIT.h"
 #include "llvm/Support/ErrorHandling.h"
 #include "llvm/System/DynamicLibrary.h"
