diff -urN a/lib/Target/ARM/ARMConstantPoolValue.h b/lib/Target/ARM/ARMConstantPoolValue.h
--- a/lib/Target/ARM/ARMConstantPoolValue.h	2010-06-15 16:50:42.000000000 +0200
+++ b/lib/Target/ARM/ARMConstantPoolValue.h	2013-09-02 20:29:54.380189599 +0200
@@ -49,10 +49,10 @@
 public:
   ARMConstantPoolValue(const Constant *cval, unsigned id,
                        ARMCP::ARMCPKind Kind = ARMCP::CPValue,
-                       unsigned char PCAdj = 0, const char *Modifier = NULL,
+                       unsigned char PCAdj = 0, const char *Modifier = 0,
                        bool AddCurrentAddress = false);
   ARMConstantPoolValue(LLVMContext &C, const char *s, unsigned id,
-                       unsigned char PCAdj = 0, const char *Modifier = NULL,
+                       unsigned char PCAdj = 0, const char *Modifier = 0,
                        bool AddCurrentAddress = false);
   ARMConstantPoolValue(const GlobalValue *GV, const char *Modifier);
   ARMConstantPoolValue();
@@ -62,7 +62,7 @@
   const char *getSymbol() const { return S; }
   const BlockAddress *getBlockAddress() const;
   const char *getModifier() const { return Modifier; }
-  bool hasModifier() const { return Modifier != NULL; }
+  bool hasModifier() const { return Modifier != 0; }
   bool mustAddCurrentAddress() const { return AddCurrentAddress; }
   unsigned getLabelId() const { return LabelId; }
   unsigned char getPCAdjustment() const { return PCAdjust; }
