diff -urN a/ld/pass1.c b/ld/pass1.c
--- a/ld/pass1.c	2013-04-02 23:44:57.451301534 +0100
+++ b/ld/pass1.c	2013-04-02 23:44:57.639299170 +0100
@@ -508,7 +508,7 @@
 	    pass1_object(file_name, file_addr, file_size, base_name, FALSE,
 			 FALSE, bundle_loader, force_weak);
 	}
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && (HAVE_DECL_VM_MSYNC)
 	vm_msync(mach_task_self(), (vm_address_t)file_addr,
 		 (vm_size_t)file_size, VM_SYNC_DEACTIVATE);
 #endif /* VM_SYNC_DEACTIVATE */
diff -urN a/ld/pass2.c b/ld/pass2.c
--- a/ld/pass2.c	2013-04-02 23:44:57.451301534 +0100
+++ b/ld/pass2.c	2013-04-02 23:44:57.639299170 +0100
@@ -269,7 +269,7 @@
 		    }
 		}
 		output_local_symbols();
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && (HAVE_DECL_VM_MSYNC)
 		vm_msync(mach_task_self(), (vm_address_t)cur_obj->obj_addr,
 			 (vm_size_t)cur_obj->obj_size, VM_SYNC_DEACTIVATE);
 #endif /* VM_SYNC_DEACTIVATE */
diff -urN a/misc/libtool.c b/misc/libtool.c
--- a/misc/libtool.c	2013-04-02 23:44:57.443301636 +0100
+++ b/misc/libtool.c	2013-04-02 23:44:57.631299271 +0100
@@ -2559,7 +2559,7 @@
 		}
 		memcpy(p, arch->members[j].object_addr,
 		       arch->members[j].object_size);
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && (HAVE_DECL_VM_MSYNC)
 		vm_msync(mach_task_self(),
 			 (vm_address_t)arch->members[j].object_addr,
 			 (vm_size_t)arch->members[j].object_size,
