diff -urN a/libstuff/macosx_deployment_target.c b/libstuff/macosx_deployment_target.c
--- a/libstuff/macosx_deployment_target.c	2013-01-07 08:19:34.041102008 +0000
+++ b/libstuff/macosx_deployment_target.c	2013-01-07 08:19:34.141050750 +0000
@@ -26,6 +26,9 @@
 #include <string.h>
 #include <mach/mach.h>
 #include <sys/types.h>
+#if defined(__unused) && defined(__linux__)
+#undef __unused
+#endif
 #include <sys/sysctl.h>
 #include "stuff/errors.h"
 #include "stuff/allocate.h"
