diff -urN a/misc/redo_prebinding.c b/misc/redo_prebinding.c
--- a/misc/redo_prebinding.c	2013-06-04 22:01:55.607831418 +0100
+++ b/misc/redo_prebinding.c	2013-06-04 22:01:55.827833415 +0100
@@ -9818,7 +9818,7 @@
 	errors++;
 }
 #endif /* defined(LIBRARY_API) */
-
+#if HAVE_GETATTRLIST
 #include <sys/attr.h>
 /*
  * Structure defining what's returned from getattrlist.  It returns all the
@@ -9903,3 +9903,13 @@
 
 	return(FALSE);
 }
+#else
+static
+enum bool
+has_resource_fork(
+                char *filename)
+{
+  return FALSE;
+}
+#endif
+
