diff -urN a/ld64/src/ld/parsers/libunwind/AddressSpace.hpp b/ld64/src/ld/parsers/libunwind/AddressSpace.hpp
--- a/ld64/src/ld/parsers/libunwind/AddressSpace.hpp	2013-11-18 23:05:36.653503853 +0000
+++ b/ld64/src/ld/parsers/libunwind/AddressSpace.hpp	2013-11-18 23:05:36.953500082 +0000
@@ -86,7 +86,7 @@
 {
 public:
 	
-	#if __LP64__
+	#if __LP64__ || defined(__x86_64__)
 		typedef uint64_t	pint_t;
 		typedef  int64_t	sint_t;
 	#else
@@ -113,7 +113,7 @@
 
 inline uintptr_t LocalAddressSpace::getP(pint_t addr)
 {
-#if __LP64__
+#if __LP64__ || defined(__x86_64__)
 	return get64(addr);
 #else
 	return get32(addr);
