diff -urN a/include/llvm-c/Disassembler.h b/include/llvm-c/Disassembler.h
--- a/include/llvm-c/Disassembler.h	2012-03-21 03:54:29.000000000 +0000
+++ b/include/llvm-c/Disassembler.h	2013-09-08 22:11:43.973437300 +0100
@@ -47,7 +47,7 @@
  * TagType for that Triple.  If symbolic information is returned the function
  * returns 1, otherwise it returns 0.
  */
-typedef int (*LLVMOpInfoCallback)(void *DisInfo, uint64_t PC,
+typedef int (*LLVMOpInfoCallback)(void *DisInfo, uint64_t pc,
                                   uint64_t Offset, uint64_t Size,
                                   int TagType, void *TagBuf);
 
@@ -161,7 +161,7 @@
  * no valid instruction.
  */
 size_t LLVMDisasmInstruction(LLVMDisasmContextRef DC, uint8_t *Bytes,
-                             uint64_t BytesSize, uint64_t PC,
+                             uint64_t BytesSize, uint64_t pc,
                              char *OutString, size_t OutStringSize);
 
 /**
