diff -urN a/autoconf/m4/link_options.m4 b/autoconf/m4/link_options.m4
--- a/autoconf/m4/link_options.m4	2010-08-12 01:53:59.000000000 +0200
+++ b/autoconf/m4/link_options.m4	2013-11-06 15:35:01.868201485 +0100
@@ -6,7 +6,7 @@
 AC_DEFUN([AC_LINK_GET_VERSION],
   [AC_CACHE_CHECK([for linker version],[llvm_cv_link_version],
   [
-   version_string="$(ld -v 2>&1 | head -1)"
+   version_string="$(${program_prefix}ld -v 2>&1 | head -1)"
 
    # Check for ld64.
    if (echo "$version_string" | grep -q "ld64"); then
diff -urN a/configure b/configure
--- a/configure	2011-10-16 04:52:56.000000000 +0200
+++ b/configure	2013-11-06 15:35:01.872201221 +0100
@@ -7524,7 +7524,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-   version_string="$(ld -v 2>&1 | head -1)"
+   version_string="$(${program_prefix}ld -v 2>&1 | head -1)"
 
    # Check for ld64.
    if (echo "$version_string" | grep -q "ld64"); then
