diff -urN a/configure.in b/configure.in
--- a/configure.in	2013-06-04 14:58:32.831711080 +0100
+++ b/configure.in	2013-06-04 18:06:04.000000000 +0100
@@ -1054,6 +1054,7 @@
   CXX=${CXX-${host_alias}-c++}
   CXXFLAGS=${CXXFLAGS-"-g -O2"}
   CC_FOR_BUILD=${CC_FOR_BUILD-gcc}
+  CXX_FOR_BUILD=${CC_FOR_BUILD-c++}
 
 else
   # Set reasonable default values for some tools even if not Canadian.
@@ -1065,7 +1066,9 @@
   # This is all going to change when we autoconfiscate...
 
   CC_FOR_BUILD="\$(CC)"
+  CXX_FOR_BUILD="\$(CXX)"
   AC_PROG_CC
+  AC_PROG_CXX
 
   # We must set the default linker to the linker used by gcc for the correct
   # operation of libtool.  If LD is not defined and we are using gcc, try to
@@ -2251,6 +2254,7 @@
 
 # Build tools.
 AC_SUBST(CC_FOR_BUILD)
+AC_SUBST(CXX_FOR_BUILD)
 AC_SUBST(config_shell)
 
 # Generate default definitions for YACC, M4, LEX and other programs that run
diff -urN a/Makefile.in b/Makefile.in
--- a/Makefile.in	2013-06-04 14:58:32.931711081 +0100
+++ b/Makefile.in	2013-06-04 14:58:33.155044415 +0100
@@ -242,7 +242,7 @@
 CC_FOR_BUILD = @CC_FOR_BUILD@
 CFLAGS_FOR_BUILD = @CFLAGS_FOR_BUILD@
 
-CXX_FOR_BUILD = $(CXX)
+CXX_FOR_BUILD = @CXX_FOR_BUILD@
 
 # Special variables passed down in EXTRA_GCC_FLAGS.  They are defined
 # here so that they can be overridden by Makefile fragments.
