diff -urN a/gcc/Makefile.in b/gcc/Makefile.in
--- a/gcc/Makefile.in	2013-07-22 21:05:52.675440600 +0100
+++ b/gcc/Makefile.in	2013-07-22 23:31:35.480500500 +0100
@@ -4113,8 +4113,12 @@
 	-rm -f $(DESTDIR)$(bindir)/$(GCC_INSTALL_NAME)$(exeext)
 	-$(INSTALL_PROGRAM) xgcc$(exeext) $(DESTDIR)$(bindir)/$(GCC_INSTALL_NAME)$(exeext)
 	-rm -f $(DESTDIR)$(bindir)/$(target_noncanonical)-gcc-$(version)$(exeext)
-	-rm -f $(DESTDIR)$(libsubdir)/libstdc++.dylib
-	$(LN_S) /usr/lib/libstdc++.6.dylib $(DESTDIR)$(libsubdir)/libstdc++.dylib
+	# Only overwrite libstdc++.dylib if specified DESTDIR, otherwise we're messing with
+	# the native Darwin system or placing .dylib files in /usr/lib on Linux or Windows.
+	-if [ ! "$(DESTDIR)" = "" -a -f $(DESTDIR)$(libsubdir)/libstdc++.dylib ]; then \
+	  rm -f $(DESTDIR)$(libsubdir)/libstdc++.dylib; \
+	  $(LN_S) /usr/lib/libstdc++.6.dylib $(DESTDIR)$(libsubdir)/libstdc++.dylib; \
+	else true; fi
 	-( cd $(DESTDIR)$(bindir) && \
 	   $(LN) $(GCC_INSTALL_NAME)$(exeext) $(target_noncanonical)-gcc-$(version)$(exeext) )
 	-if [ -f gcc-cross$(exeext) ] ; then \
